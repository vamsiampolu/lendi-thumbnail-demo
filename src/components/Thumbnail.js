import React from 'react'
import { Card, Image, Button } from 'semantic-ui-react'
import { Link } from 'react-router'

export default function Thumbnail(props) {
  console.log(`The url is ${props.url}`)
  return (<Card>
    <Image src={props.thumbnailUrl} />
    <Card.Content>
      <Card.Header>
        {props.title}
      </Card.Header>
      <Card.Meta>
        Album {props.albumId}
      </Card.Meta>
    </Card.Content>
    <Card.Content extra>
      <Button basic color='green'>
        <Link to={'/images/' + props.id}> View Image </Link>
      </Button>
    </Card.Content>
  </Card>)
}
