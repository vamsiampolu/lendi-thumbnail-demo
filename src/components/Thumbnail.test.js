import React from 'react'
import renderer from 'react-test-renderer'
import Thumbnail from './Thumbnail'

describe('Thumbnail', () => {
  it('renders correctly', function() {
    const props = {
      "albumId": 1,
      "id": 1,
      "title": "accusamus beatae ad facilis cum similique qui sunt",
      "url": "http://placehold.it/600/92c952",
      "thumbnailUrl": "http://placehold.it/150/30ac17"
    }
    const component = renderer.create(<Thumbnail {...props}/>)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
