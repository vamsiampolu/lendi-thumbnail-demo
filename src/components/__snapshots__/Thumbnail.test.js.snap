exports[`Thumbnail renders correctly 1`] = `
<div
  className="ui card"
  href={undefined}
  onClick={[Function]}>
  <img
    alt={undefined}
    className="ui image"
    height={undefined}
    src="http://placehold.it/150/30ac17"
    width={undefined} />
  <div
    className="content">
    <div
      className="header">
      accusamus beatae ad facilis cum similique qui sunt
    </div>
    <div
      className="meta">
      Album 
      1
    </div>
  </div>
  <div
    className="extra content">
    <button
      className="ui green basic button"
      onClick={[Function]}
      tabIndex={undefined}>
      <a
        onClick={[Function]}
        style={Object {}}>
         View Image 
      </a>
    </button>
  </div>
</div>
`;
